openeo-geotrellis/src/main/scala/org/openeo/geotrellis/package.scala

```diff
@@ -158,28 +160,67 @@ package object geotrellis {
     val seconds, days, undefined = Value
   }
 
-  def withRetries[R](action: => scalaj.http.HttpResponse[R]): scalaj.http.HttpResponse[R] = {
-    @tailrec
-    def attempt(retries: Int, delay: (Long, TimeUnit))(action: => scalaj.http.HttpResponse[R]): scalaj.http.HttpResponse[R] = {
-      val (amount, timeUnit) = delay
-
-      val ret = action
-      if (ret.code == 429) {
-        logger.warn(f"Got ${ret.code}. Will retry request: " + ret)
-        // TODO: Could use Retry-After header.
-        // It is not returned by https://services.terrascope.be/ at the moment.
-        // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After
-      } else {
-        return ret
-      }
-      timeUnit.sleep(amount)
-      attempt(retries - 1, (amount * 2, timeUnit)) {
-        action
-      }
+
+  /**
+   * taken from DefaultProcessApi
+   */
+  def withRetryAfterRetries[R](context: String)(fn: => HttpResponse[R])(implicit logger: Logger): HttpResponse[R] = {
+    var lastResponse: Option[HttpResponse[R]] = None
+
+    val retryable: Throwable => Boolean = {
+      case HttpStatusException(400, _, responseBody) if responseBody.contains("Request body should be non-empty.") => true
+      case HttpStatusException(statusCode, _, responseBody) if statusCode >= 500
+        && !responseBody.contains("newLimit > capacity") && !responseBody.contains("Illegal request to https") => true
```

Comment:
Also SHub-specific, just retry on every 500 response:

```suggestion
      case HttpStatusException(statusCode, _, responseBody) if statusCode >= 500 => true
```